drivers/SmartThings/zigbee-switch/src/aqara/init.lua

```diff
@@ -70,9 +112,36 @@ local function is_aqara_products(opts, driver, device)
   return false
 end
 
-local function emit_power_consumption_report_event(device, value)
-  local raw_value = value.value -- 'Wh'
+local function private_mode_handler(driver, device, value, zb_rx)
+  device:set_field(PRIVATE_MODE, value.value, { persist = true })
+end
+
+local function on_off_handler(driver, device, value, zb_rx)
+  device:emit_event_for_endpoint(
+    zb_rx.address_header.src_endpoint.value,
+    value.value and capabilities.switch.switch.on() or capabilities.switch.switch.off()
+  )
+
+  local private_mode = device:get_field(PRIVATE_MODE) or 0
+  if private_mode == 1 then
+    -- read power meter
+    device:send(AnalogInput.attributes.PresentValue:read(device):to_endpoint(POWER_METER_ENDPOINT))
+  end
+end
+
+local function wireless_switch_handler(driver, device, value, zb_rx)
+  local endpoint = wireless_switch_endpoint_map[zb_rx.address_header.src_endpoint.value]
+  local event = wireless_switch_button_event_map[value.value]
```

Comment:
Is the plan to extend this at some point?

Reply:
(beyond the one value it has now)

Reply:
No. Only switch 1, 2 and 3.

Reply:
then it seems strange to use a map in only one place with only one value

Reply:
Are you suggesting using an if-else conditional inside a function?

Reply:
I updated code. Please review it. Thanks. 😀